KenticoOnboardingApplication/src/KenticoOnboardingApplication.Api/Utils/UnityResolver.cs

```
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Http.Dependencies;
+using Unity;
+using Unity.Exceptions;
+
+namespace KenticoOnboardingApplication.Api.Utils
+{
+    internal sealed class UnityResolver : IDependencyResolver
+    {
+        private readonly IUnityContainer _container;
+
+        public UnityResolver(IUnityContainer container)
+        {
+            _container = container ?? throw new ArgumentNullException(nameof(container));
+        }
+
+        public object GetService(Type serviceType)
+        {
+            try
+            {
+                return _container.Resolve(serviceType);
+            }
+            catch (ResolutionFailedException)
+            {
+                return null;
```

This implementation is a bit misleading. 

Name the exception's variable and debug the code. You will find that not all types that are requested by GetService callees are registered (you registered to be precise) and surely guess what they are for (and that don't need to register them 😉).

Why I point this out is that you usually want your code to [fail fast](https://dzone.com/articles/fail-fast-principle-in-software-development). While you can test whether registered all contracts, it does not level with a run-time check (after all, tests mustn't be always run before deploying code (usually an accident) or might something (one cannot afford to write tests to test their tests... And who would write tests to test that tests 😆)

---

Once you solve the puzzle above, take a notice that very similar code can be found just a method above – let's try to make the code not repeat itself.